pkg/service/repair/service_repair_integration_test.go

```diff
@@ -483,6 +529,109 @@ func TestServiceGetTargetIntegration(t *testing.T) {
 	}
 }
 
+func TestServiceRepairOneJobPerHostIntegration(t *testing.T) {
+	session := CreateScyllaManagerDBSession(t)
+	h := newRepairTestHelper(t, session, repair.DefaultConfig())
+	clusterSession := CreateSessionAndDropAllKeyspaces(context.Background(), t, h.Client(), ManagedClusterHosts())
+
+	const (
+		ks1           = "test_repair_rf_1"
+		ks2           = "test_repair_rf_2"
+		ks3           = "test_repair_rf_3"
+		t1            = "test_table_1"
+		t2            = "test_table_2"
+		maxJobsOnHost = 1
+	)
+	createKeyspaceWithRF(t, clusterSession, ks1, 1, 1)
+	createKeyspaceWithRF(t, clusterSession, ks2, 2, 2)
+	createKeyspaceWithRF(t, clusterSession, ks3, 3, 3)
+	WriteData(t, clusterSession, ks1, 5, t1, t2)
+	WriteData(t, clusterSession, ks2, 5, t1, t2)
+	WriteData(t, clusterSession, ks3, 5, t1, t2)
+
+	t.Run("repair schedules only one job per host at any given time", func(t *testing.T) {
+		ctx := context.Background()
+
+		target := h.generateTarget(map[string]any{
+			"fail_fast":               true,
+			"single_host_parallelism": 1,
+		})
+
+		// The amount of currently executed repair jobs on host
+		jobsPerHost := make(map[string]int)
+		muJPH := sync.Mutex{}
+
+		// Set of hosts used for given repair job
+		hostsInJob := make(map[string][]string)
+		muHIJ := sync.Mutex{}
+
+		cnt := atomic.Int64{}
+
+		// Repair request
+		h.Hrt().SetInterceptor(httpx.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			if repairEndpointRegexp.MatchString(req.URL.Path) && req.Method == http.MethodPost {
+				hosts := strings.Split(req.URL.Query()["hosts"][0], ",")
+				muJPH.Lock()
+				defer muJPH.Unlock()
+
+				for _, host := range hosts {
+					jobsPerHost[host]++
+					if jobsPerHost[host] > maxJobsOnHost {
+						cnt.Add(1)
+						return nil, nil
+					}
+				}
+			}
+
+			return nil, nil
+		}))
+
+		h.Hrt().SetRespNotifier(func(resp *http.Response, err error) {
+			if resp == nil {
+				return
+			}
+
+			var copiedBody bytes.Buffer
+			tee := io.TeeReader(resp.Body, &copiedBody)
+			body, _ := io.ReadAll(tee)
+			resp.Body = io.NopCloser(&copiedBody)
+
+			// Response to repair schedule
+			if repairEndpointRegexp.MatchString(resp.Request.URL.Path) && resp.Request.Method == http.MethodPost {
+				muHIJ.Lock()
+				hostsInJob[string(body)] = strings.Split(resp.Request.URL.Query()["hosts"][0], ",")
+				muHIJ.Unlock()
+			}
+
+			// Response to repair status
+			if repairEndpointRegexp.MatchString(resp.Request.URL.Path) && resp.Request.Method == http.MethodGet {
+				status := string(body)
+				if status == "\"SUCCESSFUL\"" || status == "\"FAILED\"" {
+					muHIJ.Lock()
+					hosts := hostsInJob[resp.Request.URL.Query()["id"][0]]
+					muHIJ.Unlock()
+
+					muJPH.Lock()
+					defer muJPH.Unlock()
+
+					for _, host := range hosts {
+						jobsPerHost[host]--
+					}
+				}
+			}
```

Comment:
Wasn't this test flaky at the beginning? What has changed?

Reply:
it was, and actually I found the reason after couple of debugging hours...
repair_job_id is given per master host, not per cluster